src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -67,6 +64,211 @@ case class Task(
 }
 
 object Task {
+  private[this] def newHealthStatus(
```

Comment:
Nit-pick: This method really seems to be a detail implementation.

I would move it further down so that the more important things come first. Nevertheless a short description would be nice.